### PR TITLE
RFC: Bump the minimum numpy version, drop older python versions

### PR DESCRIPTION
**Pull request by ev-br**
For scipy 0.18:
- require numpy >= 1.7.1, test on 1.7.2
- drop pythons 2.6, 3.2 and 3.3 from CI

See email thread starting at https://mail.scipy.org/pipermail/scipy-dev/2015-December/021124.html for discussion.

### CODE REVIEW

**Comment by rgommers** [comment]
+1 for the dropped numpy and Python versions.

**Comment by rgommers** [comment]
The build failure looks real.

**Comment by argriffing** [comment]
+1 for dropped versions. Copying a comment I accidentally added to a different scipy fork:
```
Should this be 
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
instead?
The build failure log is
https://s3.amazonaws.com/archive.travis-ci.org/jobs/104462828/log.txt
```
I don't know if this change would be correct.

**Comment by rgommers** [comment]
@argriffing are you commenting on this line:https://github.com/scipy/scipy/blob/master/scipy/_build_utils/__init__.py#L13 or on something else?

File: scipy/spatial/ckdtree/src/query.cxx [hunk 1]
```diff
@@ -1,10 +1,5 @@
-/*
- * This would break SciPy with NumPy 1.6 so just accept the compiler
- * warning for now.
- * #define NPY_NO_DEPRECATED_API NPY_1_9_API_VERSION 
- *
- */
- 
+#define NPY_NO_DEPRECATED_API NPY_1_9_API_VERSION
```
Comment (argriffing) [inline scipy/spatial/ckdtree/src/query.cxx]: Is this correct?

**Comment by argriffing** [comment]
> are you commenting on this 

@rgommers I've added an inline comment in this PR. Maybe it is related to this issue https://github.com/numpy/numpy/issues/3008?

File: scipy/spatial/ckdtree/src/query.cxx (hunk 1 above)
Comment (rgommers) [inline scipy/spatial/ckdtree/src/query.cxx]: No, that looks wrong to me. Should be done in the spatial setup.py with ``numpy_nodepr_api``. Example: https://github.com/scipy/scipy/blob/master/scipy/optimize/setup.py#L20

IIRC the reason was that before numpy 1.10.0 the numpy deprecation macro did something quite unwanted. It's a while ago, but the detail should be in the PR where I added ``numpy_nodepr_api``.

Comment (larsmans) [inline scipy/spatial/ckdtree/src/query.cxx]: My mistake. @ev-br can you undo this one? I'm afraid I don't seem to fully understand what these macros do.

Comment (ev-br) [inline scipy/spatial/ckdtree/src/query.cxx]: Me neither. I tried to switch to using `numpy_nodepr_api` in spatial/setup.py, but that only lead to more mysterious CI failures. Reverted it now.

**Comment by ev-br** [comment]
Reverted the change in `query.cxx`, and CI is all green.

**Comment by rgommers** [comment]
LGTM now, merged. Thanks Evgeni.

**Pull request merged**